File: app/src/main/java/com/parabola/newtone/ui/view/WrapContentHeightViewPager.java
package com.parabola.newtone.ui.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.viewpager.widget.ViewPager;

/**
 * ViewPager для отображения в {@link androidx.fragment.app.DialogFragment}.<p>
 * Высчитывает свою высоту на основе внутреннего {@link View} с максимальной высотой
 */
public class WrapContentHeightViewPager extends ViewPager {

    public WrapContentHeightViewPager(Context context) {
        super(context);
    }

    public WrapContentHeightViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        boolean wrapHeight = MeasureSpec.getMode(heightMeasureSpec) == MeasureSpec.AT_MOST;
        if (wrapHeight) {
            for (int i = 0; i < getChildCount(); i++) {
                View child = getChildAt(i);
                if (child != null) {
                    child.measure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED));
                    int h = child.getMeasuredHeight();

                    heightMeasureSpec = Math.max(heightMeasureSpec, MeasureSpec.makeMeasureSpec(h, MeasureSpec.EXACTLY));
                }
            }
        }

        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    }

}

File: ui-kit/src/main/java/com/parabola/newtone/uikit/WrapContentHeightViewPager.kt
package com.parabola.newtone.uikit

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.fragment.app.DialogFragment
import androidx.viewpager.widget.ViewPager
import kotlin.math.max

/**
 * ViewPager для отображения в [DialogFragment].<p>
 * Высчитывает свою высоту на основе внутреннего [View] с максимальной высотой
 */
class WrapContentHeightViewPager(
    context: Context,
    attrs: AttributeSet?,
) : ViewPager(context, attrs) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        var resultHeightMeasureSpec = heightMeasureSpec
        val wrapHeight = MeasureSpec.getMode(resultHeightMeasureSpec) == MeasureSpec.AT_MOST
        if (wrapHeight) {
            for (i in 0 until childCount) {
                val child = getChildAt(i)
                if (child != null) {
                    child.measure(
                        widthMeasureSpec,
                        MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED),
                    )
                    val height = child.measuredHeight
                    resultHeightMeasureSpec = max(
                        resultHeightMeasureSpec,
                        MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY),
                    )
                }
            }
        }
        super.onMeasure(widthMeasureSpec, resultHeightMeasureSpec)
    }


}
